pkg/app/piped/cloudprovider/cloudrun/diff_test.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2022 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cloudrun
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDiff(t *testing.T) {
+	old, err := loadServiceManifest("testdata/old_manifest.yaml")
```

Comment:
To be honest, I'd prefer letting `loadServiceManifest` receive `io/fs/FS` and using filesystem-independent testdata.
That being said, a plenty of functions inside this project call `os.ReadFile`, hence it's fine to keep convention as is.

Reply:
I see. Is the main purpose of it a readability issue?

Reply:
It's because you can clearly identify the test data for each test case without adding test data files to the project tree that you don't know where they are referenced from.

Reply:
That make sense!
I'll refactor it all together in a separate PR since that fix will affect existing functions.